### PR TITLE
feat: better non-exported type diagnostic

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -1271,3 +1289,114 @@ fn definition_location(
...
+  /// Symbol identifiers that are not exported, but are referenced
+  /// by exported symbols.
+  non_exported_public_ids: HashSet<UniqueSymbolId>,
+}
+
+impl SymbolVisibility {
+  pub fn build(
+    graph: &ModuleGraph,
+    root_symbol: &deno_graph::type_tracer::RootSymbol,
+  ) -> Result<Self, DocError> {
```
Comment (dsherret) [inline src/parser.rs:1304]: I realized while working on this that the "type tracing" feature in deno_graph is somewhat useless and no longer used. We should refactor it to a general purpose symbol graph: https://github.com/denoland/deno_graph/issues/316

File: src/parser.rs [hunk 2]
```diff
@@ -1061,7 +1015,26 @@ impl<'a> DocParser<'a> {
...
+            let mut diagnostics = diagnostics.borrow_mut();
+            for dep in deps {
+              let module =
+                self.root_symbol.get_module_from_id(dep.module_id).unwrap();
+              let symbol = module.symbol(dep.symbol_id).unwrap();
+              diagnostics
+                .add_private_type_in_public(doc, doc_id, module, symbol);
+            }
+          }
+        }
```
Comment (bartlomieju) [inline src/parser.rs:1034]: Considering use `let/else` here?

File: src/parser.rs (hunk 1 above)
Comment (bartlomieju) [reply src/parser.rs:1304]: Agreed

**Review by bartlomieju — APPROVED** [review]

File: src/parser.rs (hunk 2 above)
Comment (dsherret) [reply src/parser.rs:1034]: I'll extract it to a method and do so.

**Pull request merged**